front/components/assistant_builder/AssistantBuilderTablesModal.tsx

```diff
@@ -155,20 +155,24 @@ const PickTable = ({
     workspaceId: owner.sId,
     dataSourceName: dataSource.name,
   });
+  const [query, setQuery] = useState<string>("");
 
   const tablesToDisplay = tables.filter(
     (t) =>
       !tablesQueryConfiguration?.[
         `${owner.sId}/${dataSource.name}/${t.table_id}`
       ]
   );
+  const filtered = tablesToDisplay.filter((t) => {
```

Comment:
Let's wrap this in a `useMemo` since the component also takes other props.

Reply:
ok